js: eshost-support related fixes

IdanHo opened the pull request.

linusg commented on Userland/Utilities/js.cpp:
```diff
@@ -738,7 +738,8 @@ static void print_value(JS::Value value, HashTable<JS::Object*>& seen_objects)
         if (is<JS::Date>(object))
             return print_date(object, seen_objects);
-        if (is<JS::Error>(object))
+        auto prototype_or_error = object.internal_get_prototype_of();
+        if (is<JS::Error>(object) || (prototype_or_error.has_value() && prototype_or_error.value() == object.global_object().error_prototype()))
```
`print_object` at the end might crash if we call it with a stored exception - also, can we only do the prototype lookup for non-`JS::Error` objects?

IdanHo replied:
ugh, I keep forgetting that we still throw normal exceptions even with ThrowCompletionOr, fixed.

ADKaster commented:
Should we add an install rule for `js` in CMake lists? I noticed running some stuff this weekend that ninja install on the Lagom directory doesn't put the utilities in lagom-install/bin

IdanHo commented:
> Should we add an install rule for `js` in CMake lists? I noticed running some stuff this weekend that ninja install on the Lagom directory doesn't put the utilities in lagom-install/bin

That would probably be nice, yeah

linusg commented:
Code LGTM, but almost forgot about the man page :)
https://github.com/SerenityOS/serenity/blob/master/Base/usr/share/man/man1/js.md

IdanHo commented:
> Code LGTM, but almost forgot about the man page :) [`master`/Base/usr/share/man/man1/js.md](https://github.com/SerenityOS/serenity/blob/master/Base/usr/share/man/man1/js.md?rgh-link-date=2021-11-24T14%3A41%3A58Z)

Whoops! Fixed. (and also added a commit that adds a couple of other js options that were already missing)
